src/walk.rs

```diff
@@ -79,11 +84,13 @@ impl<'a> Walk<'a> {
         }
     }
 
-    pub fn run(&self) -> (u64, Vec<Error>) {
+    pub fn run(&self) -> (u64, u64, u64, Vec<Error>) {
```

Comment:
I would really prefer if we do not return three `u64` here where it's impossible to tell which is which (from the signature). Instead, we should probably introduce a small struct:
``` rust
struct WalkResult {
  pub size_in_bytes: u64,
  pub file_count: u64,
  pub directory_count: u64
}
```

Reply:
Agreed.  Will add this!